### PR TITLE
Enhancements to Operator Configuration and Installation Options

### PR DESCRIPTION
**Pull request by dwertent**
### Overview
 
1. **Extended Configuration for Paladin/Besu Workloads**  
   Added support for the following fields in the operator configuration to enhance flexibility when deploying Paladin/Besu workloads:
   - `labels`
   - `annotations`
   - `envs`
   - `tolerations`
   - `affinity`
   - `nodeSelector`
   - `securityContext`

2. **New Installation Mode**  
   Introduced a configurable installation mode, allowing the operator to be installed without the default CRs. This supports environments where only the operator setup is required.

3. **Flux Integration Example**  
   Added an example configuration for using Flux.

### CODE REVIEW

File: operator/charts/paladin-operator/values.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
-
-
-# Install CRDs as part of the chart installation
-# When this is set to false, the CRDs are expected to be installed separately
-installCRDs: false
+# Installation mode. This setting determines which Custom Resources (CRs) will be installed by default when deploying this chart.
+# Supported modes:
+# - dev-net: Installs a default Paladin network (3 nodes) along with the related Smart Contracts.
+# - none (or left empty): Only the operator will be installed.
+mode: dev-net
```
Comment (onelapahead) [inline operator/charts/paladin-operator/values.yaml:5]: nit: `devnet` I think. Matches how blockchain communities refer to their broad networks as `testnet` and `mainnet` often in config.

File: operator/charts/paladin-operator/values.yaml [hunk 2]
```diff
@@ -93,13 +93,22 @@ paladin:
...
   image:
     repository: postgres
     tag: latest
-    pullPolicy: Always
\ No newline at end of file
+    pullPolicy: Always
+
+# Install CRDs as part of the chart installation
+# When this is set to false, the CRDs are expected to be installed separately
+installCRDs: false
```
Comment (onelapahead) [inline operator/charts/paladin-operator/values.yaml:114]: 📦

File: operator/charts/paladin-operator/assets/config.json [hunk 3]
```diff
@@ -2,19 +2,30 @@
   "paladin": {
     "image": "{{ .Values.paladin.image.repository }}:{{ .Values.paladin.image.tag }}",
     "imagePullPolicy": "{{ .Values.paladin.image.pullPolicy }}",
-    "labels": {
-      "app": "paladin"
-    }
+    "labels": {{ toJson .Values.paladin.labels }},
```
Comment (onelapahead) [inline operator/charts/paladin-operator/assets/config.json:5]: To be clear I think these need to be treated as _extra_ labels. The Operator should always use/prescribe a set of well-known labels i.e. `app.kubernetes.io/name` and `app.kubernetes.io/component` that it uses for consistent labeling for Service, Deployment, PDB, and NetworkPolicy selectors.

**Review by onelapahead — APPROVED** [review]
LGTM overall, thank you for all the tests, examples, and new config options 📈

File: operator/charts/paladin-operator/values.yaml (hunk 2 above)
Comment (dwertent) [reply operator/charts/paladin-operator/values.yaml:114]: Well there is still the option of having a single installation for both the CRDs and the oprtator

File: operator/charts/paladin-operator/assets/config.json (hunk 3 above)
Comment (dwertent) [reply operator/charts/paladin-operator/assets/config.json:5]: I will add that code in the operator

Comment (dwertent) [reply operator/charts/paladin-operator/assets/config.json:5]: Added to the operator: https://github.com/LF-Decentralized-Trust-labs/paladin/pull/430/commits/321eb116d991b1a25a0c7c5e9ea93d870f1c436d#diff-9d910189a9a7a42fd5edc8ea7da4b1b89fd7f019dfc98ee19e30b3d124e9da12R348

**Review by hosie — APPROVED** [review]
This is great.  Thank you.

**Pull request merged**